frontend/src/app/pages/exercises/exercise/shared/exercise-map/utility/resize-rectangle-interaction.ts

```diff
@@ -0,0 +1,156 @@
+            feature,
+            currentScale: { x: 1, y: 1 },
+        };
+        return true;
+    }
+
+    private _handleDragEvent(event: MapBrowserEvent<any>): boolean {
+        if (this.currentResizeValues === undefined) {
+            return false;
+        }
+        const mouseCoordinate = event.coordinate;
+        const newXScale =
+            (mouseCoordinate[0]! - this.currentResizeValues.originCorner[0]!) /
+            (this.currentResizeValues.draggedCorner[0]! -
+                this.currentResizeValues.originCorner[0]!);
+        const newYScale =
+            (this.currentResizeValues.originCorner[1]! - mouseCoordinate[1]!) /
+            (this.currentResizeValues.originCorner[1]! -
+                this.currentResizeValues.draggedCorner[1]!);
+        this.currentResizeValues.feature
+            .getGeometry()!
+            .scale(
+                newXScale / this.currentResizeValues.currentScale!.x,
+                newYScale / this.currentResizeValues.currentScale!.y,
+                this.currentResizeValues.originCorner
+            );
+        this.currentResizeValues.currentScale = { x: newXScale, y: newYScale };
+        return true;
+    }
+
+    private _handleUpEvent(event: MapBrowserEvent<any>): boolean {
+        if (this.currentResizeValues === undefined) {
+            return true;
+        }
+
+        const coordinates = this.currentResizeValues.feature
+            .getGeometry()!
+            .getCoordinates()!;
+        const topLeftCoordinate = coordinates.reduce<Coordinate>(
+            (smallestCoordinate, coordinate) =>
+                coordinate[0]! <= smallestCoordinate[0]! ||
+                coordinate[1]! >= smallestCoordinate[1]!
+                    ? coordinate
+                    : smallestCoordinate,
+            [Number.POSITIVE_INFINITY, Number.NEGATIVE_INFINITY]
+        );
+        this.currentResizeValues.feature.dispatchEvent(
+            new ResizeEvent(
+                this.currentResizeValues.currentScale,
+                this.currentResizeValues.originCorner,
+                topLeftCoordinate
+            )
+        );
+        this.currentResizeValues = undefined;
+        return false;
+    }
+
+    static onResize(
+        feature: Feature<LineString>,
+        callback: (event: ResizeEvent) => void
+    ) {
+        feature.addEventListener(
+            resizeRectangleEventType,
+            callback as (event: BaseEvent | Event) => void
+        );
+    }
+}
+
+interface CurrentResizeValues {
+    draggedCorner: Coordinate;
+    /**
+     * The corner that doesn't move during the resize.
+     */
+    originCorner: Coordinate;
+    /**
+     * The feature that is currently being resized.
+     */
+    feature: Feature<LineString>;
+    currentScale: { x: number; y: number };
+}
+
+// TODO: This doesn't work as a static member of ResizeEvent, because it is undefined at runtime. Why?
+const resizeRectangleEventType = 'resizerectangle';
```

Comment:
No idea. My only guess would be that something with a conflicting name already exists in the type hierarchy.